### DISCUSSION TITLE
Wifi components stolen from an enemy sub don't work

### DISCUSSION THREAD

**Discussion by FredrikM97** [FakeFishGames/Barotrauma#13778] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Wifi component within a circuit box can only communicate with another wifi component within a circuit box. 

Not working scenarios:
Wifi component -> wifi component within circuit box 
Wifi component within circuit box -> Wifi component

Working scenarios:
Wifi component -> Wifi component 
Wifi component within circuit box -> Wifi component within circuit box

### Reproduction steps

In multiplayer try send signal on wifi component outside a circuit box into the circuit box wifi component.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.3.0.3

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I unfortunately could not reproduce this. Tested with a very simple setup where a signal from a button was sent from a wifi component to another wifi component inside a circuit box, and then passed to a text display, and it seems to work with no issues:

https://github.com/FakeFishGames/Barotrauma/assets/14351070/044b8d75-c59e-41ee-80b3-701793293266

I think it could have something to do with the specific circuit you've built, could you give us some more details?

**Reply by FredrikM97** [reply]
Hi!

I played on my friends server som I did not host it. I have noticed that hosting and joining a server creates different results in multiplayer. Like circuits act bugged for the host but not user and vise versa.

The circuit I tested was a lever to wifi and then wifi inside circuit box to memory (check value) and output and the output to a display. The display and memory remained empty. Used channel 1234.

I also see that you can do "add label". I do not have that function.

I can attempt to create a video to show the issues. Any recommendation on program?

**Reply by FredrikM97** [reply]
I remember now that I stole the wifi components from another sub that I invaded. I will try to dismantle the wifi components and test if that could be the cause.

**Reply by Regalis11** [reply]
Ahh, I think that explains it! The wifi components in different submarines cannot communicate with each other (otherwise the devices in an enemy sub might for example interfere with devices in your sub, or vice versa). It's not at all obvious why they don't work if you steal them though, I think we could make them work after they've been taken into your own sub.

**Reply by FredrikM97** [reply]
That makes sense. I speculated that enemy sub have a different communication. 

It would be good if wifi did not work at all if taken from another ship to avoid the confusion with wifi inside circuit box and outside circuit box. 

 Maybe require to deconstruct to use? Fun exploit to sink enemy sub if they have wifi connected to pumps.
